### PR TITLE
Fix errors with Spotbugs 4.4.x

### PR DESCRIPTION
**Pull request by damiencarol**
Some data changed with version 4.4.0, the parser was only compatible with version 4.2.x

## Work done
- [x] improve robustness of the parser with missing attributes
- [x] add fix from #4292
- [x] fix type of CWE
- [x] add more unit tests

### CODE REVIEW

**Review by valentijnscholten — APPROVED** [review]

File: dojo/tools/spotbugs/parser.py [hunk 1]
```diff
@@ -86,41 +82,52 @@ def get_findings(self, filename, test):
             for message in bug.itertext():
                 desc += message + '\n'
 
-            dupe_key = bug.get('instanceHash')
-
-            title = bug.find('ShortMessage').text
+            shortmessage_extract = bug.find('ShortMessage')
+            if shortmessage_extract:
+                title = shortmessage_extract.getText()
```
Comment (StefanFl) [inline dojo/tools/spotbugs/parser.py:87]: Something isn't working with the title in my test. Will investigate tomorrow, no time today anymore. For now just want to make sure the PR is not merged.

**Review by StefanFl — CHANGES_REQUESTED** [review]

File: dojo/tools/spotbugs/parser.py (hunk 1 above)
Comment (Maffooch) [reply dojo/tools/spotbugs/parser.py:87]: The title in my testing appears to be fine. The new scan (4.4.0) in this PR does not use `ShortMessage`, but the[ other sample scan ](https://github.com/DefectDojo/django-DefectDojo/blob/master/dojo/unittests/scans/spotbugs/many_findings.xml)does. Here's a screenshot of both versions titles.

version 4.4.0
image

Version 4.2.x
image

Comment (StefanFl) [reply dojo/tools/spotbugs/parser.py:87]: In the version 4.2 example you see the problem. The title of the first one should be **Potential Command Injection** which is the content of `ShortMessage`.  

It works for me when I change the detection of the title to:

```
shortmessage_extract = bug.find('ShortMessage')
if shortmessage_extract is not None:
    title = bug.find('ShortMessage').text
else:
    title = bug.get('type')
```

**Review by StefanFl — CHANGES_REQUESTED** [review]

**Comment by StefanFl** [comment]
I tested the SpotBugs scan on a project in my company. I get the additional information with SpotBugs 4.4 as well. The report doesn't look very much different from SpotBugs 4.2. Nevertheless it makes sense to make the SpotBugs import more robust.

File: dojo/tools/spotbugs/parser.py (hunk 1 above)
Comment (StefanFl) [reply dojo/tools/spotbugs/parser.py:87]: The unit tests for the old version of the reports doesn't have a test for the title. With such a test you would have noticed the title has changed with this fix to a poorer version.

**Comment by damiencarol** [comment]
@StefanFl how did you tested it? with the jar alone? I tested it with Maven and the dictionary were missing leading to the bugs reported.

**Comment by StefanFl** [comment]
I use Maven as well together with the findsecbugs plugin.  This is my configuration:

```
  <plugin>
	<groupId>com.github.spotbugs</groupId>
	<artifactId>spotbugs-maven-plugin</artifactId>
	<version>${version.spotbugs.maven}</version>
			<configuration>
				<effort>Max</effort>
				<threshold>Low</threshold>
				<failOnError>true</failOnError>
				<includeFilterFile>spotbugs-include-filter.xml</includeFilterFile>
				<excludeFilterFile>spotbugs-exclude-filter.xml</excludeFilterFile>
				<plugins>
					<plugin>
						<groupId>com.h3xstream.findsecbugs</groupId>
						<artifactId>findsecbugs-plugin</artifactId>
						<version>1.10.1</version>
					</plugin>
				</plugins>
			</configuration>
	<dependencies>
	<!-- Overwrite dependency on SpotBugs if you want to specify the version of SpotBugs.
		SpotBugs itself is frequently several versions ahead of the spotbugs-maven-plugin -->
		<dependency>
			<groupId>com.github.spotbugs</groupId>
			<artifactId>spotbugs</artifactId>
			<version>${version.spotbugs}</version>
		</dependency>
	</dependencies>
</plugin>
```

where `version.spotbugs` is 4.4.0

**Comment by damiencarol** [comment]
@StefanFl found the error with title and fixed it. Made more unit tests and fixes.

File: dojo/tools/spotbugs/parser.py (hunk 1 above)
Comment (damiencarol) [reply dojo/tools/spotbugs/parser.py]: I added unit tests for title and saw the pb. You were right @StefanFl , I fixed my PR

**Review by StefanFl — APPROVED** [review]

**Pull request merged**